dev/NavigationView/NavigationView.cpp

```diff
@@ -4700,6 +4735,65 @@ bool NavigationView::IsFullScreenOrTabletMode()
     return isFullScreenMode || isTabletMode;
 }
 
+void NavigationView::SetDropShadow()
+{
+    const auto shadowCaster = m_shadowCaster.get();
+    const auto displayMode = DisplayMode();
+
+    if (displayMode == winrt::NavigationViewDisplayMode::Compact || displayMode == winrt::NavigationViewDisplayMode::Minimal)
+    {
+        if (shadowCaster)
+        {
+            const auto rootSplitView = m_rootSplitView.get();
+            const auto rootSplitViewActualWidth = rootSplitView.ActualWidth();
+
+            // If the screen real estate is too small, the left pane of SplitView ends up being smaller than OpenPaneLength and takes the width of RootSplitView.
+            // This requires a different storyboard animation than default.
+            if (rootSplitViewActualWidth < rootSplitView.OpenPaneLength())
+            {
+                // Since the width of the NavigationView pane is now different than OpenPaneLength, we need to update its TranslateX animation value to a reflect that.
+                const auto negativeSplitViewWidthMinusCompactLength = (rootSplitViewActualWidth - rootSplitView.CompactPaneLength()) * -1;
+                GetTemplateSettings()->NegativeSplitViewWidthMinusCompactLength(negativeSplitViewWidthMinusCompactLength);
+
+                if (const auto shadowCasterSmallPaneEaseInStoryboard = m_shadowCasterSmallPaneEaseInStoryboard.get())
+                {
+                    shadowCasterSmallPaneEaseInStoryboard.Begin();
+                }
+            }
+            else
+            {
+                if (const auto shadowCasterEaseInStoryboard = m_shadowCasterEaseInStoryboard.get())
+                {
+                    shadowCasterEaseInStoryboard.Begin();
```

Comment:
>shadowCasterEaseInStoryboard [](start = 20, length = 28)

Why is it necissary to start these storyboards from code behind. instead can we add the storyboard to a VSM state and just call go to state?

Reply:
Many of the animations and visual states are controlled by SplitView visual states so NavView doesn't have access to it. The workaround is to run the animations from codebehind.